subprojects/plugins/src/test/groovy/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponentTest.groovy

```diff
@@ -16,22 +16,22 @@
 
 package org.gradle.jvm.component.internal
 
+import org.gradle.api.GradleException
 import org.gradle.api.attributes.Bundling
 import org.gradle.api.attributes.Category
 import org.gradle.api.attributes.DocsType
 import org.gradle.api.attributes.Usage
 import org.gradle.api.internal.tasks.JvmConstants
 import org.gradle.api.plugins.JavaBasePlugin
 import org.gradle.api.plugins.JavaPluginExtension
-import org.gradle.api.plugins.jvm.internal.DefaultJvmFeature
-import org.gradle.api.reflect.ObjectInstantiationException
 import org.gradle.api.tasks.SourceSet
+import org.gradle.jvm.component.SingleTargetJvmFeature
 import org.gradle.test.fixtures.AbstractProjectBuilderSpec
 
 /**
  * Tests {@link DefaultJvmSoftwareComponent}.
  *
- * TODO: This tests a lot of the functionality of the {@link DefaultJvmFeature}. We should probably
+ * TODO: This tests a lot of the functionality of the {@link DefaultSingleTargetJvmFeature}. We should probably
```

Comment:
I think this is going to need some time in the wild to gather feedback before we change add multi-target support and would favor just moving it now.

Reply:
Yeah this is an older comment from when I still thought multi-target would be a higher priority. I'll update these tests.

Reply:
I've created an additional `DefaultSingleTargetJvmFeatureTest` and separated the related tests between that and `DefaultJvmSoftwareComponentTest`. Also added a number of new tests verifying new functionality.